src/evm/protocol/uniswap_v4/state.rs

```diff
@@ -279,6 +279,79 @@ impl UniswapV4State {
             sqrt_price_next
         }
     }
+
+    fn find_limits_experimentally(
+        &self,
+        token_in: Bytes,
+        token_out: Bytes,
+    ) -> Result<(BigUint, BigUint), SimulationError> {
+        // Create dummy token objects with proper addresses. This is fine since `get_amount_out`
+        // only uses the token addresses.
+        let token_in_obj =
+            Token::new(&token_in, "TOKEN_IN", 18, 0, &[Some(10_000)], Default::default(), 100);
+        let token_out_obj =
+            Token::new(&token_out, "TOKEN_OUT", 18, 0, &[Some(10_000)], Default::default(), 100);
+
+        // Find limit for token_in -> token_out
+        let max_amount_in = self.find_max_amount(&token_in_obj, &token_out_obj)?;
+        // Find limit for token_out -> token_in
+        let max_amount_out = self.find_max_amount(&token_out_obj, &token_in_obj)?;
+
+        Ok((max_amount_in, max_amount_out))
+    }
+
+    /// Finds max amount by performing exponential search.
```

Comment:
Could you add more context here on why exponential search is being used? I think the paragraph you have in the PR description would be very helpful here.